deepspeed/runtime/utils.py

```diff
@@ -703,7 +703,7 @@ def full(self, device=None):
         dist.all_gather(partition_tensors, partition_tensors[self.rank], group=self.group)
 
         for i in range(len(partition_tensors)):
-            partition_tensors[i].data = torch.zeros(1)
+            partition_tensors[i].data = torch.zeros(1, device=partition_tensors[i].data.device)
```

Comment:
You'll want to run formatting on this file, or just add spaces around the ` = `.

Reply:
@loadams thanks for pointing me to the formatting failure, it appears to be because of the redundant spaces on other device argument in this commit.